dapplet/src/index.ts

```diff
@@ -151,121 +145,214 @@ export default class TwitterFeature {
+      const connectedAccounts = await Core.connectedAccounts.getConnectedAccounts(username, 'twitter');
+      const walletForAutoclaim = await this.tippingService.getWalletForAutoclaim(
+        username + '/' + websiteName.toLowerCase(),
+      );
+      const pendingRequestsIds = await Core.connectedAccounts.getPendingRequests();
+      let pendingRequest = null;
+      let madeRequestId = -1;
+      if (!walletForAutoclaim && pendingRequestsIds && pendingRequestsIds.length) {
+        const requests = await Promise.all(
+          pendingRequestsIds.map((pendingRequest) => Core.connectedAccounts.getVerificationRequest(pendingRequest)),
+        );
+        const gId = username + '/' + websiteName.toLowerCase();
+        requests.forEach((request, i) => {
+          if (request.firstAccount === gId || request.secondAccount === gId) {
+            pendingRequest = request;
+            madeRequestId = pendingRequestsIds[i];
+          }
+        });
+      }
+      if (walletForAutoclaim) {
+        me.hidden = false;
         me.label = 'Unlink';
-        me.tooltip = `Unlink ${nearAccount} account from NEAR wallet`;
+        me.tooltip = `Disable autoclaim from @${username} to ${walletForAutoclaim} NEAR wallet.`;
+        me.hidden = false;
+      } else if (madeRequestId !== -1 && pendingRequest) {
+        if (this._isWaitForCAPendingVRequest) {
+          return;
+        } else {
+          this._isWaitForCAPendingVRequest = true;
+          me.label = 'Waiting...';
+          me.hidden = false;
+          const requestStatus = await this.waitForCAVerificationRequestResolve(madeRequestId);
+          this._isWaitForCAPendingVRequest = false;
+          alert(
+            (pendingRequest.isUnlink ? 'Disconnection of ' : 'Connection of ') +
+              pendingRequest.firstAccount.split('/')[0] +
+              ' and ' +
+              pendingRequest.secondAccount.split('/')[0] +
+              ' has been ' +
+              requestStatus,
+          );
+          this.executeInitWidgetFunctions();
+        }
+      } else if (connectedAccounts?.[0]?.[0]) {
+        me.hidden = true;
+      } else {
+        me.label = 'Link';
+        me.tooltip = `Link ${
+          parsedNearAccount ? parsedNearAccount + ' ' : ''
+        }account to @${username} using Connected Accounts smart contract`;
+        me.hidden = false;
       }
-      me.hidden = false;
     } else {
       me.hidden = true;
     }
   };
 
-  onProfileButtonLinkExec = async (profile, me) => {
-    const nearAccount = this.parseNearId(profile.authorFullname, this._network);
+  onProfileButtonLinkExec = async (profile: { id: string; authorFullname: string }, me) => {
+    const parsiedNearAccount = this.parseNearId(profile.authorFullname, this._network);
+    const { websiteName, fullname, img } = await this.getCurrentUserAsync();
+    const websiteNameLowerCase = websiteName.toLowerCase();
     try {
+      const accountGId = profile.id + '/' + websiteName.toLowerCase();
```

Comment:
Can this logic be moved to a separate service to make the dapplet code cleaner?

Reply:
fixed